src/shared/nuclearnet/nuclearnet_proxy_parser.ts

```diff
@@ -0,0 +1,101 @@
+import * as Emitter from 'component-emitter'
+import { NUClearNetPacket } from 'nuclearnet.js'
+import { NUClearNetPeer } from 'nuclearnet.js'
+
+import { Packet } from './nuclearnet_proxy_parser_socketio'
+import { TYPES } from './nuclearnet_proxy_parser_socketio'
+
+export class Encoder {
```

Comment:
This needs a lot of comments that justify its existence and what problem all this solves.